### PR TITLE
Add option for namespace package

### PR DESCRIPTION
**Pull request by jl-wynen**

### CODE REVIEW

**Review by SimonHeybrock — COMMENTED** [review]
Not sure I read the diff correctly, but I do not understand the change: Shouldn't the namespace be joined with the projectname in many places? Or, it may be simpler to set `projectname=esssans` and `namespace=ess` and split `projectname` in relevant places?

**Comment by jl-wynen** [comment]
> Or, it may be simpler to set projectname=esssans and namespace=ess and split projectname in relevant places?

That was my idea. Or do we want it as a naming convention that the namespace is a prefix?

**Comment by SimonHeybrock** [comment]
> Not sure I read the diff correctly, but I do not understand the change: Shouldn't the namespace be joined with the projectname in many places? Or, it may be simpler to set `projectname=esssans` and `namespace=ess` and split `projectname` in relevant places?

I think I understand now, it is being joined in the folder name, right?

**Comment by jl-wynen** [comment]
Not joined. There is a separate folder. So namespace=ess, projectname=sans leads to
```
src/ess/sans/__init__.py
```
I.e., the user-entered project name does not get modified. So in the case of esssans, we need namespace=ess, projectname=sans

**Comment by SimonHeybrock** [comment]
Ok, all clear after looking at it locally, GitHub does not do a great job rendering this.

Can you improve the new help message to make it clear that the namespace must(?) be the first part of `projectname`?

**Comment by jl-wynen** [comment]
> Can you improve the new help message to make it clear that the namespace must(?) be the first part of projectname?

Id doesn't have to be. This is at best a convention.

File: src/{% if namespace_package %}{{namespace_package}}{% endif %}/{{projectname}}/__init__.py.jinja
[the remark this answers is not in the document]
Comment (jl-wynen) [reply src/{% if namespace_package %}{{namespace_package}}{% endif %}/{{projectname}}/__init__.py.jinja:1]: Correct. Changed it

**Comment by SimonHeybrock** [comment]
Did you try this out somewhere? For example, `esssans`, which we want to transition anyway?

**Comment by jl-wynen** [comment]
Not in an already existing package. I doubt this will work without manual fixes. But I did try it when creating a new package.

File: src/{% if not namespace_package %}{{projectname}}{% endif %}/py.typed
Comment (SimonHeybrock) [inline src/{% if not namespace_package %}{{projectname}}{% endif %}/py.typed:1]: How does copier/jinja handle the `if` here? `namespace_package` is not `bool` but `str`. Is an empty string false?

**Comment by jl-wynen** [comment]
Forgot about the changes to the build system. This PR depends on #48

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**